packages/js/components/src/rich-text-editor/rich-text-editor.tsx

```diff
@@ -75,13 +91,19 @@ export const RichTextEditor: React.VFC< RichTextEditorProps > = ( {
 						// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 						// @ts-ignore This property was recently added in the block editor data store.
 						__experimentalClearBlockSelection: false,
+						mediaUpload,
 					} }
 					onInput={ debounceChange }
 					onChange={ debounceChange }
 				>
 					<ShortcutProvider>
-						<EditorWritingFlow />
+						<EditorWritingFlow
+							blocks={ blocksRef.current }
```

Comment:
I don't love passing this down as opposed to getting it from the store, but there's a delay initially when setting blocks in the store that causes the editor to think it's empty when it's not.